src/tiled/propertybrowser.cpp

```diff
@@ -857,6 +835,55 @@ QtVariantProperty *PropertyBrowser::createProperty(PropertyId id, int type,
     return property;
 }
 
+void PropertyBrowser::addNewProperties(Object* object) {
+    mUpdating = true;
+    // Add the built-in properties for each object type
+    switch (object->typeId()) {
+    case Object::MapType:
+        addMapProperties();
+        break;
+    case Object::MapObjectType:
+        addMapObjectProperties();
+        break;
+    case Object::LayerType:
+        switch (static_cast<Layer*>(object)->layerType()) {
+        case Layer::TileLayerType:
+            addTileLayerProperties();
+            break;
+        case Layer::ObjectGroupType:
+            addObjectGroupProperties();
+            break;
+        case Layer::ImageLayerType:
+            addImageLayerProperties();
+            break;
+        }
+        break;
+    case Object::TilesetType:
+        addTilesetProperties();
+        break;
+    case Object::TileType:
+        addTileProperties();
+        break;
+    case Object::TerrainType:
+        addTerrainProperties();
+        break;
+    }
```

Comment:
Please keep the original formatting for this whole switch block. I had it nicely aligned.